doc/releases/release-notes-3.5.rst

```diff
@@ -20,6 +20,9 @@ API Changes
 Changes in this release
 =======================
 
+* The kernel :c:func:`k_mem_slab_free` function has changed its signature, now
```

Comment:
My main concern is that this one liner in the release notes is not enough for someone using Zephyr to know the nature and impact of the change and the fact that they HAVE to change their code or risk their application changing behavior or crashing. IMO we need to isolate breaking API changes into a seperate document with more details and information about how migration to new API can be done (assisted by scripts, etc.)

Reply:
Instead of a separate document, how about a new section in the existing document, at the top, that says "Things you need to change in your code to migrate to this version"?

Reply:
Actually, I think you are right. Let's just introduce a new migration document for each release. I will send a Pull Request for this.